### PR TITLE
Initialise youtube player on client-side

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This change initialises youtube players on the client-side rather than on the server-side.  This has the advantage that consent for personalised ads is available at the point where the player is initialised and so can be passed through to the player embed call.

I've only done the bare minimum to ensure players continue to work.
So the work to avoid loading players until they are needed is still to be done.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/atoms/youtube.spec.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    trackYoutubeEvents: jest.fn(() => null),
+}));
+
+describe('youtube', () => {
+    it('adds player to page', () => {
+        if (document.body) {
+            document.body.innerHTML =
+                '<div data-media-atom-id="atomId" class="u-responsive-ratio u-responsive-ratio--hd youtube-media-atom">' +
+                '<div id="youtube-assetId"></div>' +
+                '</div>';
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/atoms/youtube.spec.js]: Could probably get some value using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) here

File: static/src/javascripts/projects/common/modules/atoms/youtube-player.js [hunk 2]
```diff
@@ -32,28 +35,42 @@ type Handlers = {
 const onPlayerStateChangeEvent = (
     event,
     handlers: Handlers,
-    el: HTMLElement
+    el: ?HTMLElement
 ) => {
+    if (el && config.page.isDev) {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/atoms/youtube-player.js]: Prefer `config.get('page.isDev')` over direct access to properties on `config`

**Review by akash1810 — COMMENTED** [review]
I probably missing something obvious, but I can't see where we're instantiating the iframe anymore...

File: static/src/javascripts/projects/common/modules/atoms/youtube.spec.js (hunk 1 above)
Comment (kelvin-chappell) [reply static/src/javascripts/projects/common/modules/atoms/youtube.spec.js]: Yes that's a good idea.  I'll do it!

File: static/src/javascripts/projects/common/modules/atoms/youtube-player.js (hunk 2 above)
Comment (kelvin-chappell) [reply static/src/javascripts/projects/common/modules/atoms/youtube-player.js]: Thanks.  I'll change this too

**Comment by kelvin-chappell** [comment]
@akash1810 The iframe is magically instantiated here: https://github.com/guardian/frontend/blob/49b24ffa1aedeadb8dbcebb59542b472bf261c2e/static/src/javascripts/projects/common/modules/atoms/youtube-player.js#L91-L105
This converts the placeholder div into the player iframe.

**Comment by kelvin-chappell** [comment]
I haven't tested this on Code yet, but I've tested it locally on `/money/video/2018/may/25/football-tickets-how-resale-sites-rip-off-fans-video` and `/politics/2017/jun/02/tory-attack-ad-corbyn-remarks-context`

**Review by akash1810 — APPROVED** [review]

**Review by SiAdcock — APPROVED** [review]
Thanks @kelvin-chappell!

**Comment by gk2go** [comment]
How do you guys do it, have a video overlay for fast loading and still have autoplay work on mobile (after clicking on the overlay)? My question on https://stackoverflow.com/questions/51702308/how-to-fast-load-youtube-embeds-keeping-1-click-play-on-mobile-phones-also?noredirect=1#comment90366076_51702308

**Pull request merged**